src/upload/mod.rs

```diff
@@ -263,6 +265,75 @@ pub async fn upload_package_to_anaconda(
     Ok(())
 }
 
+/// Uploads a package to a channel in an S3 bucket.
+pub async fn upload_package_to_s3(
+    storage: &AuthenticationStorage,
+    channel: Url,
+    endpoint_url: Option<Url>,
+    region: Option<String>,
+    force_path_style: Option<bool>,
+    package_files: &Vec<PathBuf>,
+) -> miette::Result<()> {
+    // If either endpoint_url or region is set, ensure that both are set
+    if endpoint_url.is_some() != region.is_some() {
+        return Err(miette::miette!(
+            "Both endpoint_url and region must be specified together"
+        ));
+    }
+    // TODO: Add CLI args for SECRET_ACCESS_KEY, ACCESS_KEY_ID, and SESSION_TOKEN
```

Comment:
@pavelzw do you think it's necessary to implement these arguments? To me it feels unnecessary to introduce a new process how to pass authentication information.

Reply:
given we already support non-persisted authentication using `AWS_SECRET_ACCESS_KEY` and `AWS_ACCESS_KEY_ID`, it's not strictly needed tbf

but it's maybe a bit awkward that this is the only upload subcommand that doesn't support credentials via CLI flag